tests/phpunit/unit/MediaWiki/Request/EditRequestSaverTest.php

```diff
@@ -0,0 +1,206 @@
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\Tests\Unit\MediaWiki\Request;
+
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequest;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestSaver;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ItemReconciler;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ReconciledItem;
+use MediaWiki\Extension\WikibaseReconcileEdit\Reconciliation\ReconciliationServiceItem;
+use MediaWiki\Extension\WikibaseReconcileEdit\Wikibase\FluidItem;
+use PHPUnit\Framework\TestCase;
+use Status;
+use User;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\Lib\Store\EntityRevision;
+use Wikibase\Repo\EditEntity\EditEntity;
+use Wikibase\Repo\EditEntity\MediawikiEditEntityFactory;
+
+/**
+ * @covers \MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestSaver
+ *
+ * @license GPL-2.0-or-later
+ */
+class EditRequestSaverTest extends TestCase {
+
+	/** @var PropertyId */
+	private $reconcileProperty;
+
+	/** @var User */
+	private $user;
+
+	public function setUp(): void {
+		parent::setUp();
+		$this->reconcileProperty = new PropertyId( 'P1' );
+		$this->user = $this->createMock( \User::class );
+	}
+
+	public function newRequestSaver(
+		MediawikiEditEntityFactory $editEntity = null,
+		ItemReconciler $itemReconciler = null
+	) {
+		$editEntity = $editEntity ?: $this->createMock( MediawikiEditEntityFactory::class );
+		$itemReconciler = $itemReconciler ?: $this->createMock( ItemReconciler::class );
+
+		return new EditRequestSaver( $editEntity, $itemReconciler );
+	}
+
+	public function testEditRequestSaverDoesNothing_WhenNoRequests(): void {
+		$saver = $this->newRequestSaver();
+		$status = $saver->persistEdits( [], 'token', $this->user );
+		$this->assertTrue( $status->isGood() );
+	}
+
+	public function testEditRequestSavesWithout_OtherItems(): void {
+		$item = FluidItem::withId( 'Q1' )->item();
+
+		$itemReconciler = $this->createMock( ItemReconciler::class );
+		$itemReconciler->method( 'reconcileItem' )
+			->with( $item, $this->reconcileProperty )
+			->willReturn( new ReconciledItem(
+				$item,
+				new ReconciliationServiceItem( $item, false )
+			) );
+
+		$expectedRevision = new EntityRevision( $item, 1234 );
+		$editEntity = $this->createMock( EditEntity::class );
+		$editEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $expectedRevision ] ) );
+
+		$editEntityFactory = $this->createMock( MediawikiEditEntityFactory::class );
+		$editEntityFactory->method( 'newEditEntity' )
+			->with( $this->user, $item->getId(), false )
+			->willReturn( $editEntity );
+		$saver = $this->newRequestSaver( $editEntityFactory, $itemReconciler );
+
+		$status = $saver->persistEdits( [
+			new EditRequest( $this->reconcileProperty, $item )
+		], '', $this->user );
+
+		$this->assertTrue( $status->isGood() );
+		$this->assertEquals( $expectedRevision, $status->getValue()[0]['revision'] );
```

Comment:
I think this could even be `assertSame()`? We don’t expect the saver to create a new instance, right?

Reply:
done